src/app/services/courses.service.ts

```diff
@@ -1,37 +1,44 @@
-import { Injectable } from '@angular/core';
+import {Injectable} from '@angular/core';
 
-import { courses } from "../mockArrays/mockCourses";
-import { ICourse } from "../models/Course";
+import {courses} from "../mockArrays/mockCourses";
+import {ICourse} from "../models/Course";
 
 @Injectable({
   providedIn: 'root'
 })
 
 export class CoursesService {
 
+  public coursesMock: ICourse[];
+
+  constructor() {
+    this.coursesMock = courses;
+  }
+
+
   public getCourses(): ICourse[] {
-    return courses;
+    return this.coursesMock;
   }
 
-  public getCourseById(id: string): ICourse[] {
-    return courses.filter(course => course.id == +id)
+  public getCourseById(id: string): ICourse | undefined {
+    return this.coursesMock.find(course => course.id == +id)
   }
 
   public createCourse(course: ICourse): void {
     courses.push(course)
   }
 
-  public updateCourse(id: string, course: ICourse): ICourse[] {
-    return courses.map(function (currentCourse) {
+  public updateCourse(id: number, course: ICourse): void {
+    this.coursesMock.forEach(function (currentCourse) {
       if (currentCourse.id == +id) {
         currentCourse = course
       }
-      return currentCourse
     })
   }
 
   public removeCourse(id: string): void {
-    courses.splice(courses.findIndex(course => course.id === +id), 1)
-    console.log(courses)
+    console.log(this.coursesMock)
+    this.coursesMock.splice(this.coursesMock.findIndex(course => course.id == +id), 1)
```

Comment:
try use filter  this.coursesMock =  this.coursesMock.filter((course) => {
      return course.id !== +id;
    })

Reply:
done